pkg/ratelimit/ratelimit.go

```diff
@@ -30,24 +30,28 @@ func New(r time.Duration, burst int) *Limiter {
 	}
 }
 
-// Allow checks if the limiter that belongs to 'key' has not exceeded the limit.
+// Allow checks if the limiter that belongs to key has not exceeded the limit.
 func (l *Limiter) Allow(key string, count int) bool {
+	limiter := l.limiterForKey(key)
+	return limiter.AllowN(time.Now(), count)
+}
 
+// limiterForKey returns limiter used for specifed key.
+func (l *Limiter) limiterForKey(key string) *rate.Limiter {
 	l.mtx.Lock()
 	defer l.mtx.Unlock()
 
-	limiter, ok := l.limiter[key]
-	if !ok {
-		limiter = rate.NewLimiter(l.rate, l.burst)
-		l.limiter[key] = limiter
+	if limiter, exists := l.limiter[key]; exists {
+		return limiter
 	}
 
-	return limiter.AllowN(time.Now(), count)
```

Comment:
So it would be easily achieved by just doing an `l.mtx.Unlock` here after the if case instead of a different function no?

Reply:
That would have same effect as fix for overlocking.